File: Data Structures & Algorithms/binary-search/submission-5.cpp
class Solution {
public:
    int search(vector<int>& nums, int target) {
        int n = nums.size();
        int left = 0, right = nums.size()-1;
        while (right >= left){
            int mid = left + (right-left)/2;
            cout<<mid<<" ";
            if(nums[mid]==target) return mid;
            else if(nums[mid]>target) right = mid-1;
            else left = mid+1;
        }
        return -1;
    }
};

File: Data Structures & Algorithms/two-integer-sum/submission-5.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        ind = {}

        for i, n in enumerate(nums):
            ind[n] = i

        for i, n in enumerate(nums):
            diff = target - n
            if diff in ind and ind[diff] != i:
                return [i, ind[diff]]
        
        return []